lessons/git/git-collaboration-2in1/index.md

```diff
@@ -0,0 +1,291 @@
+    Author: Magdalena Dobromila <magdalena@rettigova.cz>
+    Date:   Tue Mar 28 19:27:08 2017 +0200
+
+        Přidáno mé jméno a povolání
+
+    commit 0f305972803131cb6c8637359fee8ede3005bba6
+    Merge: effa89b 175f4cc
+    Author: Oskar Hollmann <oskar@hollmann.me>
+    Date:   Tue Nov 8 20:17:00 2016 +0100
+
+        Merge pull request #1 from muzikovam/master
+        
+        Tady mas jmeno
+
+Příkaz `git log` má mnoho přepínačů, přehlednější historii změn můžeme získat pomocí `git log --oneline --graph --decorate`.
+
+### Posílání změn (`git push`)
+
+Nyní stačí požádat autora projektu, aby naše změny začlenil do repozitáře na GitHubu 
+a všichni uživatelé projektu, na kterém pracujeme, budou mít užitek z našich změn.
+Nemáme ale do původního repozitáře práva na zápis, musíme tedy poslat změny nejprve do naší kopie repozitáře,
+vizte obrázek na začátku.
+
+Udělejte si účet na GitHubu (jestli ho ještě nemáte), a pak jděte na adresu, kterou jste použili pro `git clone`. 
+Vlevo nahoře je tlačítko „Fork”, klikněte na něj. Tím se na GitHubu vytvoří vaše kopie repozitáře: 
+adresa by měla být něco jako `https://github.com/tvojejmeno/prezencka`.
+
+A teď, jak nahrát změny z našeho počítače na GitHub? Git si u každého repozitáře na našem počítači pamatuje adresy, 
+odkud se dají stahovat a kam se dají posílat změny. 
+Seznam těchto adres nám ukáže příkaz `git remote -v`. Třeba:
+
+    $ git remote -v
+    origin  https://github.com/asgeirrr/prezencka (fetch)
+    origin  https://github.com/asgeirrr/prezencka (push)
+
+Tenhle výstup znamená, že pod zkratkou *origin* se schovává adresa, ze kterého jsme repozitář naklonovali.
+
+Přidejme si podobnou zkratku pro vlastní repozitář na GitHubu. Můžeme ho pojmenovat např. *public*,
+aby nám bylo jasné, že do tohoto repozitáře můžeme nahrávat změny.
+
+    $ git remote add public https://github.com/tvojejmeno/prezencka 
+
+Zkontrolujme, že se nám to povedlo:
+
+    $ git remote -v
+    origin  git@github.com:asgeirrr/prezencka.git (fetch)
+    origin  git@github.com:asgeirrr/prezencka.git (push)
+    public  https://github.com/tvojejmeno/prezencka (fetch)
+    public  https://github.com/tvojejmeno/prezencka (push)
+
+Tolik k nastavení -- `git remote add` stačí udělat jednou pro každý repozitář. Pak už můžeme změny nahrávat pomocí: 
+
+    $ git push public master
+
+což znamená: pošli na adresu uloženou pod zkratkou `public` větev `master`.
+
+Funguje? Můžeme se podívat na `https://github.com/tvojejmeno/prezencka` v prohlížeči, a ujistit se, že tam změny jsou.
+
+
+### Žádost o začlenění (pull request)
+
+Teď zbývá požádat autory původního projektu, aby změny z tvého sdíleného repozitáře přidali do svojí kopie. 
+GitHub na to má vlastní mechanismus zvaný *pull request* (žádost o začlenění).
+
+Na stránce původního projektu (na adrese, kterou jsme použili na začátku pro `git clone`)
+by mělo být oznámení o nově nahrané větvi s velkým zeleným tlačítkem `Compare & pull request`. 
```

Comment:
Tak to nevím, co s tím, kdyžtak je budu navigovat, ale mě se to bežně ukazuje, když pushnu větev do nějakého svého forku.